arangosh/Restore/RestoreFeature.cpp

```diff
@@ -816,13 +816,32 @@ arangodb::Result restoreView(arangodb::httpclient::SimpleHttpClient& httpClient,
 arangodb::Result triggerFoxxHeal(arangodb::httpclient::SimpleHttpClient& httpClient) {
   using arangodb::Logger;
   using arangodb::httpclient::SimpleHttpResult;
-  const std::string FoxxHealUrl = "/_api/foxx/_local/heal";
-
   std::string body = "";
 
+  // check if the foxx api is available.
+  const std::string statusUrl = "/_admin/status";
   std::unique_ptr<SimpleHttpResult> response(
-      httpClient.request(arangodb::rest::RequestType::POST, FoxxHealUrl,
+      httpClient.request(arangodb::rest::RequestType::POST, statusUrl,
                          body.c_str(), body.length()));
+  auto res =  ::checkHttpResponse(httpClient, response, "check status", body);
+  if(res.fail()) {
+    res.reset(res.errorNumber(), "While checking avilablility of foxx-api for heal: " + res.errorMessage());
+    return res;
```

Comment:
IMHO we should not return here, for the following reason:
Due to security reasons, the `/_admin/status` API may be turned off entirely, and return HTTP 403.
In this case we want the restore to go on and still try to call the Foxx API for self-healing (because in fact it may still work). We should only not try to call the Foxx API when we are 100% sure it is turned off.